File: functions/nodejs-typescript-esbuild/src/functions/hello-2/handler.ts
import { middyfy } from "@libs/middyfy";
import {
  Handler,
  ValidatedScalewayHandlerEvent,
  formatScalewayHandlerJSONResponse,
} from "@libs/scalewayServerless";

import { helloWorld } from "hello-world-npm";
import { Response } from "scaleway-functions";

import { ISchema } from "./schema";

export const hello: Handler = async (
  event: ValidatedScalewayHandlerEvent<ISchema>
): Promise<Response> => {
  const data = event?.body;
  const timeToWait = 500;
  await new Promise((res) => setTimeout(res, timeToWait));

  return formatScalewayHandlerJSONResponse({
    statusCode: 200,
    headers: {
      "Content-Type": "application/json",
    },
    body: {
      message: {
        one: `Hello ${
          data?.name || "person"
        }, welcome to the exciting Serverless world!`,
        two: helloWorld(),
        env: process.env.TEST_2,
      },
    },
  });
};
export const handler = middyfy(hello); // Its important that the exported funtion name is handler.

File: functions/python-dependencies/handler.py
import requests


def handle(event, context):
    resp = requests.get("https://www.scaleway.com")
    return {
        "body": f"Response status: {resp.status_code}",
        "headers": {
            "Content-Type": ["text/plain"],
        },
    }


if __name__ == "__main__":
    from scaleway_functions_python import local

    local.serve_handler(handle)
